addon-manager/tests/src/test/java/org/jboss/forge/addon/manager/ui/ForgeUpdateDistributionCommandTest.java

```diff
@@ -0,0 +1,296 @@
+      DocumentBuilder dBuilder;
+      try
+      {
+         dBuilder = dbFactory.newDocumentBuilder();
+         String oldUiAddonVersion = "";
+         File installedXml = new File(homeAddonsDirPath + "/installed.xml");
+         Document doc = dBuilder.parse(installedXml);
+         // this does not work properly
+         Element documentElement = doc.getDocumentElement();
+         NodeList childNodes = documentElement.getElementsByTagName("addon");
+         for (int i = 0; i <= childNodes.getLength() - 1; i++)
+         {
+            Element item = (Element) childNodes.item(i);
+            if (item.getNodeName().equals("addon"))
+            {
+               String addonName = item.getAttribute("name");
+               if (addonName.equals("org.jboss.forge.addon:addon-manager"))
+               {
+                  oldUiAddonVersion = item.getAttribute("version");
+                  return oldUiAddonVersion;
+               }
+            }
+         }
+      }
+      catch (ParserConfigurationException e)
+      {
+         e.printStackTrace();
+      }
+      catch (Exception e)
+      {
+         System.out.println("Why?");
+      }
+      return null;
+   }
+
+   /**
+    * This method changes the installed.xml so it has outdated addon-manager
+    * 
+    * @param homeAddonsDirPath
+    */
+   private void makeManagerAddonVersionOlder(String homeAddonsDirPath)
+   {
+      DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+      DocumentBuilder dBuilder;
+      try
+      {
+         dBuilder = dbFactory.newDocumentBuilder();
+         File installedXml = new File(homeAddonsDirPath + "/installed.xml");
+         Document doc = dBuilder.parse(installedXml);
+         // this does not work properly
+         Element documentElement = doc.getDocumentElement();
+         NodeList childNodes = documentElement.getElementsByTagName("addon");
+         for (int i = 0; i <= childNodes.getLength() - 1; i++)
+         {
+            Element item = (Element) childNodes.item(i);
+            if (item.getNodeName().equals("addon"))
+            {
+               String addonName = item.getAttribute("name");
+               if (addonName.equals("org.jboss.forge.addon:addon-manager"))
+               {
+                  item.setAttribute("version", FORGE_OLD_VERSION);
+                  TransformerFactory transformerFactory = TransformerFactory.newInstance();
+                  Transformer transformer = transformerFactory.newTransformer();
+                  DOMSource source = new DOMSource(doc);
+                  StreamResult result = new StreamResult(installedXml);
+                  transformer.transform(source, result);
+                  return;
+               }
+            }
+         }
+      }
+      catch (ParserConfigurationException e)
+      {
+         e.printStackTrace();
+      }
+      catch (Exception e)
+      {
+         System.out.println("Why?");
```

Comment:
This is funny, but these exceptions probably should be fatal, no? Reporting that the version could not be determined? Or is this actually expected?